lib/secretFactory.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const BaseFactory = require('./baseFactory');
+const Secret = require('./secret');
+const iron = require('iron');
+const nodeify = require('./nodeify');
+const hoek = require('hoek');
+// Get symbols for private fields
+const password = Symbol();
+
+let instance;
+
+/**
+ * Seal secret value and return the secret with the sealed value
+ * @method sealSecret
+ * @param  {SecretModel}        secret  Secret to seal
+ * @param  {String}             pw      Password to seal with
+ * @return {Promise}
+ */
+function sealSecret(secret, pw) {
+    return nodeify.withContext(iron, 'seal', [secret.value, pw, iron.defaults])
+        .then(sealed => {
+            secret.value = sealed;
+
+            return secret;
+        });
+}
+
+/**
+ * Unseal secret value and return the secret with the unsealed value
+ * @param  {SecretModel}        secret  Secret to unseal
+ * @param  {String}             pw      Password to unseal with[description]
+ * @return {Promise}
+ */
+function unsealSecret(secret, pw) {
+    return nodeify.withContext(iron, 'unseal', [secret.value, pw, iron.defaults])
+        .then(unsealed => {
+            secret.value = unsealed;
+
+            return secret;
+        });
+}
+
+class SecretFactory extends BaseFactory {
+    /**
+     * Construct a SecretFactory object
+     * @method constructor
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore     Object that will perform operations on the datastore
+     * @param  {String}    config.password      Password for encryption operations
+     */
+    constructor(config) {
+        super('secret', config);
+        this[password] = config.password;
+    }
+
+    /**
+     * Instantiate a Secret class
+     * @method createClass
+     * @param  config
+     * @return {Secret}
+     */
+    createClass(config) {
+        const c = hoek.applyToDefaults(config, { password: this[password] });
+
+        return new Secret(c);
+    }
+
+    /**
+     * Create a secret model
+     * Need to seal the secret before saving
+     * @method create
+     * @param  {Object}     config
+     * @param  {String}     config.pipelineId      Pipeline Id
+     * @param  {String}     config.name            Secret name
+     * @param  {String}     config.value           Secret value
+     * @param  {Boolean}    config.allowInPR       Whether this secret can be shown in PR builds
+     * @return {Promise}
+     */
+    create(config) {
+        return sealSecret(config, this[password])
+            .then(secret => super.create(hoek.applyToDefaults(config, { value: secret.value })));
```

Comment:
I think the `hoek.applyToDefaults` is a redundant call. You already modify the `config` parameter in `sealSecret()`.